app/util/job.ts

```diff
@@ -29,13 +30,18 @@ export default async function cancelAndSaveJob(
     } else {
       ({ job } = await Job.byRequestId(tx, jobID));
     }
+
     if (job) {
       if (job.status !== JobStatus.CANCELED || !shouldIgnoreRepeats) {
         job.status = JobStatus.CANCELED;
         job.validateStatus();
         job.cancel(message);
         await job.save(tx);
-        if (shouldTerminateWorkflows) {
+        const isTurboWorkflow = await checkIfTurboWorkflow(tx, jobID, logger);
```

Comment:
Knowing that the plan is to deprecate Argo, I think you could probably rearrange this to something like the following to make it easier to find and remember to delete the Argo stuff:

```
        const updatedAt = new Date();
        await tx('work_items').where({ jobID: job.jobID }).update({ status: WorkItemStatus.CANCELED, updatedAt });

        // The following can be removed once Argo is removed
        const isArgoWorkflow = !await checkIfTurboWorkflow(tx, jobID, logger);
        if (isArgoWorkflow && shouldTerminateWorkflows) {
          await terminateWorkflows(job, logger);
        }
```